hippo.go

```diff
@@ -108,8 +108,8 @@ func (c *Client) SetProxy(url *url.URL) {
 	c.transport.Proxy = http.ProxyURL(url)
 }
 
-func (c *Client) NewTagBatchPartRequest(method string, url string, data []byte) (*http.Request, error) {
-	req, err := http.NewRequest(method, url, bytes.NewReader(data))
+func (c *Client) NewTagBatchPartRequest(method string, url string, gzippedData []byte) (*http.Request, error) {
```

Comment:
Oh jeez. The root of this is the naming of NewTagBatchPartRequest. I added the gzipping, not realizing that this function was also used for building a request for other APIs.

So, we should either add a second method, or do what you did, but rename it.

Can you rename it to something like NewGZippedAPIRequest?

Reply:
b1175de6242faf12a3e46e800a60f5c914c8a9f2

Reply:
@ingwarsw b1175de6242faf12a3e46e800a60f5c914c8a9f2 Added some tests to this code.